File: fluentcore/src/main/java/com/mcreater/fxfluent/util/SystemUtil.java
package com.mcreater.fxfluent.util;

public class SystemUtil {
    public static boolean isWindows() {
        return System.getProperty("os.name").toLowerCase().contains("windows");
    }

    public static boolean isMac() {
        return System.getProperty("os.name").toLowerCase().contains("mac") || System.getProperty("os.name").toLowerCase().contains("osx");
    }

    public static boolean isLinux() {
        return System.getProperty("os.name").toLowerCase().contains("linux");
    }
    public static void startGCLoop() {
        new Thread(() -> {
            while (true) {
                System.gc();

                try {
                    Thread.sleep(500);
                }
                catch (InterruptedException e) {

                }
            }
        }).start();
    }
}

File: fluentcore/src/main/java/com/mcreater/fxfluent/util/SystemUtil.kt
package com.mcreater.fxfluent.util

import java.util.*

class SystemUtil {
    companion object {
        @JvmStatic
        fun isWindows(): Boolean {
            return System.getProperty("os.name").lowercase(Locale.getDefault()).contains("windows")
        }

        @JvmStatic
        fun isMac(): Boolean {
            return System.getProperty("os.name").lowercase(Locale.getDefault())
                .contains("mac") || System.getProperty("os.name").lowercase(Locale.getDefault()).contains("osx")
        }

        @JvmStatic
        fun isLinux(): Boolean {
            return System.getProperty("os.name").lowercase(Locale.getDefault()).contains("linux")
        }
    }
}